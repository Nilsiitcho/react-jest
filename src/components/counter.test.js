import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Counter from './Counter';

// Garante que o componente é renderizado com o contador inicial igual a zero
test('renders counter with initial count of zero', () => {
    const { getByText } = render(<Counter />);
    const countElement = getByText(/0/i);
    expect(countElement).toBeInTheDocument();
});

// Garante que o ao clicar no botão + o contador incrementa uma unidade
test('increment button increments count', () => {
    const { getByText } = render(<Counter />);
    const incrementButton = getByText('+');
    const countElement = getByText(/0/i);

    fireEvent.click(incrementButton);

    expect(countElement).toHaveTextContent('1');
});

// Garante que o ao clicar no botão - o contador decrementa uma unidade
test('decrement button decrements count', () => {
    const { getByText } = render(<Counter />);
    const decrementButton = getByText('-');
    const countElement = getByText(/0/i);

    fireEvent.click(decrementButton);

    expect(countElement).toHaveTextContent('-1');
});

// Realiza um teste de Snapshot no componente
test('renders counter correctly', () => {
    const { asFragment } = render(<Counter />);
    expect(asFragment()).toMatchSnapshot();
});