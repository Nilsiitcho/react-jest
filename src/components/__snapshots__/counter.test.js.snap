// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders counter correctly 1`] = `
<DocumentFragment>
  <div>
    <h2>
      Contador
    </h2>
    <div>
      <button>
        -
      </button>
      <span>
        0
      </span>
      <button>
        +
      </button>
    </div>
  </div>
</DocumentFragment>
`;
